### PR TITLE
feat: build manylinux compatible builds

### LINKED ISSUE #1402: ✨[Feature] PyPi binaries
**Is your feature request related to a problem? Please describe.**
Having to use a custom index is less than ideal if I'd like to for example have this library be part of extra dependencies in my `setup.py`

**Describe the solution you'd like**
 `pip install torch-tensorrt`

### PR DESCRIPTION
**Pull request by narendasan**
# Description

Start building wheels that a manylinux compatible

Fixes #1402 

## Type of change

Please delete options that are not relevant and/or add your own.

- New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: py/ci/build_whl.sh [hunk 1]
```diff
@@ -2,40 +2,66 @@
 
 # Example usage: docker run -it -v$(pwd)/..:/workspace/TRTorch build_trtorch_wheel /bin/bash /workspace/TRTorch/py/build_whl.sh
```
Comment (peri044) [inline py/ci/build_whl.sh:3]: Replace trtorch references here

File: py/ci/build_whl.sh [hunk 2]
```diff
@@ -2,40 +2,66 @@
...
+    PY_BUILD_CODE=cp37-cp37m
+    PY_VERSION=3.7
+    PY_NAME=python${PY_VERSION}
+    PY_DIR=/opt/python/${PY_BUILD_CODE}
+    PY_PKG_DIR=${PY_DIR}/lib/${PY_NAME}/site-packages/
+    ${PY_DIR}/bin/python -m pip install --upgrade pip
+    ${PY_DIR}/bin/python -m pip install -r requirements.txt
+    ${PY_DIR}/bin/python -m pip install setuptools wheel auditwheel
+    ${PY_DIR}/bin/python setup.py bdist_wheel --release --ci
+    LD_LIBRARY_PATH=$LD_LIBRARY_PATH:${PY_PKG_DIR}/torch/lib:${PY_PKG_DIR}/tensorrt/:${CUDA_HOME}/lib64:${CUDA_HOME}/lib64/stubs ${PY_DIR}/bin/python -m auditwheel repair $(cat ${PROJECT_DIR}/py/ci/soname_excludes.params) --plat manylinux_2_17_x86_64 dist/torch_tensorrt-*-${PY_BUILD_CODE}-linux_x86_64.whl
```
Comment (peri044) [inline py/ci/build_whl.sh:22]: Can you let me know what is happening here with this soname_excludes ?

File: py/ci/build_whl.sh [hunk 3]
```diff
@@ -2,40 +2,66 @@
...
+    PY_DIR=/opt/python/${PY_BUILD_CODE}
+    PY_PKG_DIR=${PY_DIR}/lib/${PY_NAME}/site-packages/
+    ${PY_DIR}/bin/python -m pip install --upgrade pip
+    ${PY_DIR}/bin/python -m pip install -r requirements.txt
+    ${PY_DIR}/bin/python -m pip install setuptools wheel auditwheel
+    ${PY_DIR}/bin/python setup.py bdist_wheel --release --ci
+    LD_LIBRARY_PATH=$LD_LIBRARY_PATH:${PY_PKG_DIR}/torch/lib:${PY_PKG_DIR}/tensorrt/:${CUDA_HOME}/lib64:${CUDA_HOME}/lib64/stubs ${PY_DIR}/bin/python -m auditwheel repair  $(cat ${PROJECT_DIR}/py/ci/soname_excludes.params) --plat manylinux_2_17_x86_64 dist/torch_tensorrt-*-${PY_BUILD_CODE}-linux_x86_64.whl
 }
 
 #build_py311() {
```
Comment (peri044) [inline py/ci/build_whl.sh:67]: Should we remove this and add later ?

File: py/ci/build_whl.sh [hunk 4]
```diff
@@ -44,31 +70,50 @@ build_py310() {
...
-    cp ${PROJECT_DIR}/bazel-bin/libtorchtrt.tar.gz ${PROJECT_DIR}/py/dist/libtorchtrt-${TORCHTRT_VERSION}-cudnn${CUDNN_VERSION}-tensorrt${TRT_VERSION}-cuda${CUDA_VERSION}-libtorch${TORCH_VERSION}-x86_64-linux.tar.gz
+    CUDA_VERSION=$(cd ${PROJECT_DIR}/py && ${PY_DIR}/bin/python3 -c "from versions import __cuda_version__;print(__cuda_version__)")
+    TORCHTRT_VERSION=$(cd ${PROJECT_DIR}/py && ${PY_DIR}/bin/python3 -c "from versions import __version__;print(__version__)")
+    TRT_VERSION=$(cd ${PROJECT_DIR}/py && ${PY_DIR}/bin/python3 -c "from versions import __tensorrt_version__;print(__tensorrt_version__)")
+    CUDNN_VERSION=$(cd ${PROJECT_DIR}/py && ${PY_DIR}/bin/python3 -c "from versions import __cudnn_version__;print(__cudnn_version__)")
+    TORCH_VERSION=$(${PY_DIR}/bin/python -c "from torch import __version__;print(__version__.split('+')[0])")
+    cp ${PROJECT_DIR}/bazel-bin/libtorchtrt.tar.gz ${PROJECT_DIR}/py/wheelhouse/libtorchtrt-${TORCHTRT_VERSION}-cudnn${CUDNN_VERSION}-tensorrt${TRT_VERSION}-cuda${CUDA_VERSION}-libtorch${TORCH_VERSION}-x86_64-linux.tar.gz
```
Comment (peri044) [inline py/ci/build_whl.sh:90]: Where are we using this renamed libtorchtrt-XXX.tar.gz file ?

File: .circleci/config.yml [hunk 5]
```diff
@@ -768,18 +769,58 @@ jobs:
                   cd ~/project/py/
                   docker build -t torch_tensorrt_release_env --build-arg trt_version=<< pipeline.parameters.trt-version-short >> -f ci/Dockerfile.ci .
             - run:
-                name: Build Python packages and pre-cxx11-abi tarball
+                name: Build Python packages
                 command: |
                   cd ~/project/py/
                   cp ~/project/toolchains/ci_workspaces/WORKSPACE.x86_64.release.rhel ~/project/WORKSPACE
-                  docker run -it --rm -v ~/project:/workspace/project torch_tensorrt_release_env /bin/bash /workspace/project/py/ci/build_whl.sh
+                  docker run -it --rm --gpus all -v ~/project:/workspace/project torch_tensorrt_release_env /bin/bash -c "source /workspace/project/py/ci/build_whl.sh && $(circleci tests split --total=5 ~/project/py/ci/build_manifest.txt)"
```
Comment (peri044) [inline .circleci/config.yml:776]: So we build torch_tensorrt wheels for Python 3.7, 3.8, 3.9, 3.10 versions but install 3.9 whl only for testing torch-trt correct ?

File: py/ci/build_whl.sh (hunk 4 above)
Comment (narendasan) [reply py/ci/build_whl.sh:90]: thats the build of the precxx11 c++ library

File: .circleci/config.yml (hunk 5 above)
Comment (narendasan) [reply .circleci/config.yml:776]: yes for now. release testing probably should test all

**Comment by peri044** [comment]
status : 
If I don't have torch and tensorrt installed and I run pip install -i https://test.pypi.org/simple/ torch-tensorrt, I get the error 
```
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 20.3/20.3 MB 7.1 MB/s eta 0:00:00
  Downloading https://test-files.pythonhosted.org/packages/3a/1e/bbd08b7b2ef7f96cd4351054deab66457834f4e1436ccb169037e71a39e7/torch-tensorrt-0.0.0.tar.gz (8.9 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [13 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-h8cldzod/torch-tensorrt_f8be91da1be844129bb09646e54b371f/setup.py", line 125, in <module>
          raise RuntimeError(open("ERROR.txt", "r").read())
      RuntimeError:
```
If they are installed, I don't and it works

**Review by peri044 — APPROVED** [review]
Linter fails with some weird error. The changes LGTM. 
Tested with `pip install --extra-index-url https://test.pypi.org/simple/ torch-tensorrt` and it works fine.

**Pull request merged**